src/app/state/canvas/canvas.state.ts

```diff
@@ -0,0 +1,9 @@
+import { IDrawableObject } from "src/app/core/objects/object";
+import { fabric } from "fabric";
+export interface ICanvasState {    
+    canvasStates: string[];    
+}
+
+export const initialCanvasState: ICanvasState={
+    canvasStates:[]
```

Comment:
Store should ideally contain only the present objects on the canvas. Not the history of all the states of the canvas over time. Refer [Three Principles of Redux](https://redux.js.org/understanding/thinking-in-redux/three-principles) to get a few guidelines on how we should use state management.

Reply:
modified to store only the current state of the canvas.